### PR TITLE
DNN-7976 - Make changes on page editing reflect on other components in page modules

### PR DESCRIPTION
**Pull request by ricardodiasc**
This will make changes in visibility be reflected in the tree view and page details.
- Connected components that watch selectedPage with redux state
- Change the PersonaBarTreeview to watch redux state and update visibility based on selectedPage after save
- Minor code improvements

### CODE REVIEW

File: src/Modules/Content/Dnn.PersonaBar.Pages/Pages.Web/src/components/PageDetails/PageUrl/PageExternalUrl/PageExternalUrl.jsx [hunk 1]
```diff
@@ -1,4 +1,6 @@
-import React, {Component, PropTypes} from "react";
+import React, {Component} from "react";
+import PropTypes from "prop-types";
+import { connect } from "react-redux";
```
Comment (tpluscode) [inline src/Modules/Content/Dnn.PersonaBar.Pages/Pages.Web/src/components/PageDetails/PageUrl/PageExternalUrl/PageExternalUrl.jsx:3]: This is something I assume we should avoid. Is `PageExternalUrl` itself a stateless component? Isn't it an issue that we're making it directly dependent on redux?

**Review by tpluscode — COMMENTED** [review]
Please see my comment. Also, you added tests for `Breadcrumb`. Whys this one in particular? Do we want more, for other modified elements?

**Comment by ricardodiasc** [comment]
The breadcrumb came from another branch. It is already merged, so it does not relate to this commit.

**Comment by tpluscode** [comment]
fixes #560

**Pull request merged**